source/unix/opensslcrypto_ecc.c

```diff
@@ -228,15 +228,55 @@ struct aws_ecc_key_pair *aws_ecc_key_pair_new_generate_random(
     return NULL;
 }
 
+static int s_ec_key_set_private_key(EC_KEY *key, struct aws_byte_cursor priv) {
+    BIGNUM *priv_n = BN_bin2bn(priv.ptr, priv.len, NULL);
+    if (!priv_n) {
+        return AWS_ERROR_CAL_CRYPTO_OPERATION_FAILED;
+    }
+
+    if (!EC_KEY_set_private_key(key, priv_n)) {
+        BN_free(priv_n);
+        return AWS_ERROR_CAL_CRYPTO_OPERATION_FAILED;
+    }
+
+    BN_free(priv_n);
+    return AWS_OP_SUCCESS;
+}
+
+static int s_ec_key_set_public_key(EC_KEY *key, struct aws_byte_cursor pub_x, struct aws_byte_cursor pub_y) {
+    BIGNUM *pub_x_num = BN_bin2bn(pub_x.ptr, pub_x.len, NULL);
+    BIGNUM *pub_y_num = BN_bin2bn(pub_y.ptr, pub_y.len, NULL);
+
+    const EC_GROUP *group = EC_KEY_get0_group(key);
+    EC_POINT *point = EC_POINT_new(group);
+
+    if (EC_POINT_set_affine_coordinates_GFp(group, point, pub_x_num, pub_y_num, NULL) != 1) {
+        goto on_error;
+    }
+
+    if (EC_KEY_set_public_key(key, point) != 1) {
+        goto on_error;
+    }
+
+    EC_POINT_free(point);
+    BN_free(pub_x_num);
+    BN_free(pub_y_num);
+
+    return AWS_OP_SUCCESS;
+
+on_error:
+    EC_POINT_free(point);
```

Comment:
Loos like our previous logic has a null protection, but not here. 
According to Google AI summary:
```
Safe to use: You can pass NULL to BN_free() without any risk. It is a safe operation, [says the Ubuntu Manpage](https://manpages.ubuntu.com/manpages/noble/man3/BN_clear_free.3ssl.html).
```
Wow, the doc it linked seems legit.

Reply:
openssl is kinda inconsistent about those. those 3 frees are fine and can handle null (ex. https://github.com/aws/aws-lc/blob/0fed4dbc456406bdaf7158fd55d05a52d9998ce4/crypto/fipsmodule/ec/ec.c#L595 and similar in openssl).
but let me add null check just for consistency cause we usually check.